cherrysrv/client.go

```diff
@@ -16,16 +16,17 @@ const (
 
 // Client connection storing basic PC data
 type Client struct {
-	conn   *net.TCPConn // tcpsocket connection.
-	Name   string       // Name of the user.
-	Status atomic.Int32
+	conn     net.Conn // tcpsocket connection.
+	Name     string   // Name of the user.
+	Status   atomic.Int32
+	Channels []*Channel
 }
 
 func (c *Client) String() string {
 	return c.Name
 }
 
-func newClient(conn *net.TCPConn) *Client {
+func newClient(conn net.Conn) *Client {
```

Comment:
This change makes it much easier to build tests since the entire server doesn't have to be constructed.